engine/src/main/java/org/hibernate/validator/internal/engine/messageinterpolation/el/RootResolver.java

```diff
@@ -44,47 +40,33 @@
 
 	@Override
 	public Class<?> getType(ELContext context, Object base, Object property) {
-		return resolve( context, base, property ) ? Object.class : null;
+		return null;
 	}
 
 	@Override
 	public Object getValue(ELContext context, Object base, Object property) {
-		if ( resolve( context, base, property ) ) {
-			if ( !isProperty( (String) property ) ) {
-				throw new PropertyNotFoundException( "Cannot find property " + property );
-			}
-			return getProperty( (String) property );
-		}
 		return null;
 	}
 
 	@Override
 	public boolean isReadOnly(ELContext context, Object base, Object property) {
-		return false;
+		return true;
 	}
 
 	@Override
 	public void setValue(ELContext context, Object base, Object property, Object value) {
-		if ( resolve( context, base, property ) ) {
-			setProperty( (String) property, value );
-		}
+		throw new PropertyNotWritableException();
```

Comment:
AFAICS it's not something that should occur. And all the other resolvers throw a `PropertyNotWritableException` so we'd better be consistent.